### PR TITLE
Add HTTP auth check

### PR DESCRIPTION
**Pull request by drasko**
Signed-off-by: Drasko DRASKOVIC <drasko.draskovic@gmail.com>

### CODE REVIEW

File: http/api/transport.go [hunk 1]
```diff
@@ -76,14 +77,32 @@ func decodeRequest(_ context.Context, r *http.Request) (interface{}, error) {
...
 func authorize(r *http.Request) (string, error) {
 	var apiKey string
+	var id string
+
 	if apiKey = r.Header.Get("Authorization"); apiKey == "" {
 		return "", errUnauthorizedAccess
 	}
 
-	return apiKey, nil
+	identityURL := os.Getenv("HTTP_ADAPTER_MANAGER_URL") + "/identity"
```
Comment (mijicd) [inline http/api/transport.go]: This can be set at top-level, as a constant, instead of setting it per each request.

File: http/api/transport.go [hunk 2]
```diff
@@ -76,14 +77,32 @@ func decodeRequest(_ context.Context, r *http.Request) (interface{}, error) {
...
+		return "", err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Authorization", apiKey)
+
+	c := &http.Client{}
+	resp, err := c.Do(req)
+	if err != nil {
+		return "", err
```
Comment (mijicd) [inline http/api/transport.go]: Make sure this error is properly handled. The safest way is to utilize domain errors.

**Review by mijicd — CHANGES_REQUESTED** [review]
Make sure the build is fixed.

File: http/api/transport.go (hunk 2 above)
Comment (drasko) [reply http/api/transport.go]: What do you mean by this? I thought that upper layer above `transport.go` handles this error, so I just propagate it.

Comment (mijicd) [reply http/api/transport.go]: Safely ignore this one, it was deprecated by #138 .

File: http/api/transport.go [hunk 3]
```diff
@@ -76,14 +79,29 @@ func decodeRequest(_ context.Context, r *http.Request) (interface{}, error) {
...
 func authorize(r *http.Request) (string, error) {
 	var apiKey string
+	var id string
+
 	if apiKey = r.Header.Get("Authorization"); apiKey == "" {
 		return "", errUnauthorizedAccess
 	}
 
-	return apiKey, nil
+	authURL := os.Getenv("HTTP_ADAPTER_MANAGER_URL")
```
Comment (mijicd) [inline http/api/transport.go]: It's better to perform this load & fallback at main, and just pass the configured value to this module.

File: http/api/transport.go [hunk 4]
```diff
@@ -76,14 +79,29 @@ func decodeRequest(_ context.Context, r *http.Request) (interface{}, error) {
 	return msg, nil
 }
 
-// TODO: contact an auth provider
 func authorize(r *http.Request) (string, error) {
 	var apiKey string
+	var id string
```
Comment (mijicd) [inline http/api/transport.go]: Not needed, it is created and initialized below.

**Review by mijicd — CHANGES_REQUESTED** [review]

File: http/api/transport.go (hunk 3 above)
Comment (drasko) [reply http/api/transport.go]: Sure, but how to pass to this module :)

Comment (mijicd) [reply http/api/transport.go]: Store the client reference to the service itself. Modify [this signature](https://github.com/mainflux/mainflux/blob/master/cmd/http/main.go#L47) and use it as an entrypoint for injecting the client. Then, store the client reference when initializing the transport layer through `MakeHandler` method and use it to authorize requests.

**Pull request merged**